OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs

```diff
@@ -27,22 +28,29 @@ public ResupplyAircraft(Actor self)
 
 		public override Activity Tick(Actor self)
 		{
-			var host = aircraft.GetActorBelow();
-
-			if (host == null)
-				return NextActivity;
-
-			if (aircraft.IsPlane)
-				return ActivityUtils.SequenceActivities(
-					aircraft.GetResupplyActivities(host)
-					.Append(new CallFunc(() => aircraft.UnReserve()))
-					.Append(new WaitFor(() => NextActivity != null || Reservable.IsReserved(host)))
-					.Append(new TakeOff(self))
-					.Append(NextActivity).ToArray());
-
-			// If is helicopter move away as soon as the resupply ends
-			return ActivityUtils.SequenceActivities(
-				aircraft.GetResupplyActivities(host).Append(new TakeOff(self)).Append(NextActivity).ToArray());
+			if (inner == null)
```

Comment:
IMO this would be clearer as

```csharp
if (inner != null)
{
	inner = ActivityUtils.RunActivity(self, inner);
	return this;
}
// your if case with one less level of indentation
```